pingcap/tiup
# Feature/disable log or data dir overlapped

### What problem does this PR solve? 

Fix #1047 

### What is changed and how it works?


### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Manual test as below:

```bash
cat << EOF > topo.yml
global:
  user: tidb6wu
  group: pingcap
  deploy_dir: tidb1-deploy
  data_dir: tidb1-data

server_configs:
  tidb:
    binlog.enable: true
    binlog.ignore-error: false
  tikv:
    storage.reserve-space: 1K
  pump:
    storage.stop-write-at-available-space: 1 mib

pd_servers:
  - host: n3
  - host: n4
  - host: n5

tikv_servers:
  - host: n1
    port: 32160
    status_port: 32180
    log_dir: '/home/tidb6wu/tidb1-data/tikv-32160/log'
    data_dir: '/home/tidb6wu/tidb1-data/tikv-32160'
  - host: n2
  - host: n3
  EOF

tiup-cluster deploy c1 v4.0.4 topo.yml
 ``` 

The check will failed of overlapError

![image](https://user-images.githubusercontent.com/29431502/105572211-bc414500-5d90-11eb-9757-6b0a9e0dd36a.png)


Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation


Release notes:

```release-note
NONE
```
